permission_validators/account/remove_key_value/src/lib.rs

```diff
@@ -0,0 +1,56 @@
+//! Validator that checks [`RemoveKeyValue`] instruction
+//! related to accounts and respective [`Grant`] and [`Revoke`] instructions.
+
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use iroha_wasm::validator::{pass_conditions, prelude::*};
+
+/// Strongly-typed representation of `can_remove_key_value_in_user_account` permission token.
+#[derive(Token, Validate, pass_conditions::derive_conversions::account::Owner)]
+#[validate(pass_conditions::account::Owner)]
+pub struct CanRemoveKeyValueInUserAccount {
+    account_id: <Account as Identifiable>::Id,
+}
+
+/// Validate [`RemoveKeyValue`] instruction as well as [`Grant`] and [`Revoke`] instructions for
+/// [`can_remove_key_value_in_user_account`] permission tokens.
+///
+/// # [`RemoveKeyValue`]
+///
+/// ## Pass
+///
+/// - [`RemoveKeyValue`] `object_id` is not an [`AccountId`];
+/// - `authority` is the account owner;
+/// - `authority` has a corresponding [`can_remove_key_value_in_user_account`] permission token.
+///
+/// ## Deny
+///
+/// If none of the `Pass` conditions are met.
+///
+/// # [`Grant`] and [`Revoke`]
+///
+/// For more details about [`Grant`] and [`Revoke`] instructions validation,
+/// see [`can_remove_key_value_in_user_account`].
+///
+/// [`can_remove_key_value_in_user_account`]: CanRemoveKeyValueInUserAccount
+pub fn validate(authority: <Account as Identifiable>::Id, instruction: Instruction) -> Verdict {
+    validate_grant_revoke!(<CanRemoveKeyValueInUserAccount>, (authority, instruction));
+
+    let Instruction::RemoveKeyValue(remove_key_value) = instruction else {
+        pass!();
+    };
+
+    let IdBox::AccountId(account_id) = remove_key_value.object_id
+        .evaluate_on_host()
+        .dbg_expect("Failed to evaluate `RemoveKeyValue` object id") else {
+        pass!();
+    };
+
+    pass_if!(account_id == authority);
```

Comment:
maybe we can name it `assert` and/or `assert_eq`?

Reply:
Which would conflict with the built ins, and have a slightly different semantic load. These are not `debug` assertions.

Reply:
Yeah, I agree that we should not replace std assertions

Reply:
ok, I seem to have a preference for overloading